### PR TITLE
feat(go-sdk): new method get request id given nats message

### PR DESCRIPTION
**Pull request by kafkaphoenix**
## Description

As users sometimes we need to unmarshall messages like in the process trigger use case so we can get back the request ID

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other changes (ci configuration, documentation or any other kind of changes)

## Checklist:

- [ ] I have created tests for my code changes, and the tests are passing.
- [ ] I have executed the pre-commit hooks locally.
- [ ] My change requires a change to the documentation (create a new issue if the documentation has not been updated).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: go-sdk/sdk/messaging/messaging_internals.go [hunk 1]
```diff
@@ -140,3 +140,24 @@ func (ms Messaging) prepareOutputMessage(msg []byte) ([]byte, error) {
 
 	return outMsg, nil
 }
+
+func (ms Messaging) GetRequestID(data []byte) (string, error) {
```
Comment (alexrodfe) [inline go-sdk/sdk/messaging/messaging_internals.go:144]: Perhaps a "GetRequestIDFromAny" would be more suitable, it is rather a larger method name, but just to add context

**Review by alexrodfe — APPROVED** [review]

File: go-sdk/sdk/messaging/messaging_internals.go (hunk 1 above)
Comment (kafkaphoenix) [reply go-sdk/sdk/messaging/messaging_internals.go:144]: i thought about it but data already suggests the origin of the request id

Comment (dfinteligenz) [reply go-sdk/sdk/messaging/messaging_internals.go:144]: Why don't we receive a `nats.Msg` as an input parameter?
That way it's clearer that this method is for extracting the RequestID from a `nats.Msg`, not from any byte array.

File: py-sdk/sdk/sdk/messaging/messaging.py [hunk 2]
```diff
@@ -166,6 +173,26 @@ def _new_response_msg(self, payload: Any, request_id: str, msg_type: MessageType
             message_type=msg_type,
         )
 
+    def get_request_id(self, data: bytes) -> (str, Exception):
```
Comment (dfinteligenz) [inline py-sdk/sdk/sdk/messaging/messaging.py:176]: ditto

**Review by dfinteligenz — CHANGES_REQUESTED** [review]

File: go-sdk/sdk/messaging/messaging_internals.go (hunk 1 above)
Comment (kafkaphoenix) [reply go-sdk/sdk/messaging/messaging_internals.go]: updated

File: py-sdk/sdk/sdk/messaging/messaging.py (hunk 2 above)
Comment (kafkaphoenix) [reply py-sdk/sdk/sdk/messaging/messaging.py]: done

**Review by alexrodfe — APPROVED** [review]

**Review by dfinteligenz — DISMISSED** [review]

**Review by dfinteligenz — APPROVED** [review]

**Pull request merged**